Microsoft.Azure.CosmosRepository/src/Options/RepositoryOptions.cs

```diff
@@ -80,6 +81,11 @@ public class RepositoryOptions
         /// </summary>
         public RepositorySerializationOptions SerializationOptions { get; set; }
 
+        /// <summary>
+        /// The <see cref="TokenCredential"/> which can be used to access azure resources, including Cosmos DB.
+        /// </summary>
+        public TokenCredential TokenCredential { get; set; } = null;

```

Comment:
Is this properly deserialized if set in the JSON configuration provider? In other words, does this work?
```json
{
    "RepositoryOptions": {
        "TokenCredential": {
            "Token": "51AE5DDxxxxxxxxxxxxxxxx556E2E"
        }
    }
}
```

Also, @mumby0168 I though of you when I was just looking at the API ref docs on the `CosmosClient` type. They have a builder pattern object for the client itself 🤓:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.fluent.cosmosclientbuilder?view=azure-dotnet

Reply:
In the response to the first question, I am really unsure, would have to try it out, I know the pattern people usually follow is configuring this identity by using most of the time the `new DefaultAzureCredential();`. So would be cool if it did work but not sure what use case it would have?

No way we should totally add this builder to one of the samples!